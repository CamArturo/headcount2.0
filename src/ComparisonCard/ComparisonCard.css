.comparison-card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  -webkit-box-shadow: 4px 4px 5px 0px rgba(10, 10, 10, 0.65);
  -moz-box-shadow: 4px 4px 5px 0px rgba(10, 10, 10, 0.65);
  box-shadow: 4px 4px 5px 0px rgba(10, 10, 10, 0.65);
  border-radius: 8px;
  background-color: #075057;
  color: #fff;
  margin: 20px;
  text-align: center;
  padding: 1rem;
  height: 350px;
  width: 250px;
}

.comparison-name {
  text-align: center;
  font-size: 1.2rem;
  height: 76px;
  margin: 0;
  padding: 0 .5rem 0 .5rem;
  letter-spacing: 1px;
  display: flex;
  align-items: center;
}

.comparison-card-body {
  width: 100%;
  background-color: #ecf2f3;
  height: 100vh;
  border-bottom-left-radius: 5px;
  border-bottom-right-radius: 5px;
}

.comparison-card-body h3 {
  text-align: center;
  color: #9e0c09;
}

.comparison-card-body li {
  color: #9e0c09;
}

.comparison-data {
  list-style-type: none;
  width: 90%;
}