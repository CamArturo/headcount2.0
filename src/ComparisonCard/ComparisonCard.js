import React from 'react';
import './ComparisonCard.css';

const ComparisonCard = ({comparisonData}) => {
  let stats = Object.keys(comparisonData).map((item, index) => {
    return (
      <h2 key={`compare-card${index}`}>{item} : {comparisonData[item]}</h2>
    );
  });
  return (
    <div className="comparison-card">
      {stats}
    </div>
  );
};

export default ComparisonCard;