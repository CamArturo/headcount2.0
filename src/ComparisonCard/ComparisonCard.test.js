import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonCard from './ComparisonCard'
import { shallow, mount } from 'enzyme';
import '../setupTests';

describe('Comparison Card Test', () => {
  it('Should match the comparison card snapshot', () => {
    // const data = 
    const comparisonCard = shallow(<ComparisonCard 
      comparisonData={
        {
        'ADAMS COUNTY 14': 0.709,
        'COLORADO': 0.53,
        'compared': 0.748
      } 
    }
      />)

    expect(comparisonCard).toMatchSnapshot()
  })
})