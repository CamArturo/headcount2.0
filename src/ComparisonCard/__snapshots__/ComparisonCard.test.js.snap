// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comparison Card Test Should match the comparison card snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonCard
    comparisonData={
        Object {
            "ADAMS COUNTY 14": 0.709,
            "COLORADO": 0.53,
            "compared": 0.748,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ADAMS COUNTY 14
           : 
          0.709
</h2>,
        <h2>
          COLORADO
           : 
          0.53
</h2>,
        <h2>
          compared
           : 
          0.748
</h2>,
      ],
      "className": "comparison-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "compare-card0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "ADAMS COUNTY 14",
            " : ",
            0.709,
          ],
        },
        "ref": null,
        "rendered": Array [
          "ADAMS COUNTY 14",
          " : ",
          0.709,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "compare-card1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "COLORADO",
            " : ",
            0.53,
          ],
        },
        "ref": null,
        "rendered": Array [
          "COLORADO",
          " : ",
          0.53,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "compare-card2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "compared",
            " : ",
            0.748,
          ],
        },
        "ref": null,
        "rendered": Array [
          "compared",
          " : ",
          0.748,
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ADAMS COUNTY 14
             : 
            0.709
</h2>,
          <h2>
            COLORADO
             : 
            0.53
</h2>,
          <h2>
            compared
             : 
            0.748
</h2>,
        ],
        "className": "comparison-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "compare-card0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "ADAMS COUNTY 14",
              " : ",
              0.709,
            ],
          },
          "ref": null,
          "rendered": Array [
            "ADAMS COUNTY 14",
            " : ",
            0.709,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "compare-card1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO",
              " : ",
              0.53,
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO",
            " : ",
            0.53,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "compare-card2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "compared",
              " : ",
              0.748,
            ],
          },
          "ref": null,
          "rendered": Array [
            "compared",
            " : ",
            0.748,
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
